src/CoreShop/Bundle/CoreBundle/Customer/CustomerLoginService.php

```diff
@@ -36,4 +44,10 @@ public function loginCustomer(CustomerInterface $customer)
         $token = new UsernamePasswordToken($customer, null, 'coreshop_frontend', $customer->getRoles());
         $this->securityTokenStorage->setToken($token);
     }
+
+    public function logoutCustomer()
+    {
+        $this->securityTokenStorage->setToken(null);
+        $this->requestStack->getCurrentRequest()->getSession()->invalidate();
```

Comment:
I am not sure about that. Why do you even need that anyway? Couldn't you just call the logout url directly?

Reply:
I mean, that is something the symfony firewall handles

Reply:
Yeah, maybe you have right. I wanted add this method only to have consistent behaviour for this service. If we have login we expect logout also but in fact it's bit overengineer.